### PR TITLE
Eeprom

### PR DESCRIPTION
**Pull request by Kavikick**
Wrapper for the onboard EEPROM

### CODE REVIEW

File: .github/actions/spelling/expect.txt [hunk 1]
```diff
@@ -59,6 +59,7 @@ RTDIN
 RTDINLOW
 sizeof
 strlen
+TANKID
```
Comment (prestoncarman) [inline .github/actions/spelling/expect.txt]: I think you could avoid adding `TANKID` by using `TANK_ID`.

File: src/Devices/EEPROM_TC.h [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+  // accessor methods
+  double getPH();
+  double getTemp();
+  double getTankID();
+  double getCorrectedTemp();
+  double getKP();
+  double getKI();
+  double getKD();
+  double getMac();  // Not used?
```
Comment (prestoncarman) [inline src/Devices/EEPROM_TC.h]: The comment here is confusing me.

Its interesting to consider if the comment or getMac function should be included. We are storing things in EEPROM with specific address. If the function was removed, now we would have a hole in the data storage area which could be confusing. Although if we move all the functions below up, then old versions will have an issue.

Maybe for now, lets just create an issue to review if getMac() is being used.

@jgfoster Thoughts?

Comment (jgfoster) [reply src/Devices/EEPROM_TC.h]: Yes, let's create an issue "Investigate need for `getMac()`" and then replace the "Not used?" comment with "See <URL> for a discussion about this function"

**Review by jgfoster — CHANGES_REQUESTED** [review]
See notes for a couple small change requests.

File: src/Devices/EEPROM_TC.cpp [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+double EEPROM_TC::getKP() {
+  return EEPROM.read(KP_ADDRESS);
+}
+double EEPROM_TC::getKI() {
+  return EEPROM.read(KI_ADDRESS);
+}
+double EEPROM_TC::getKD() {
+  return EEPROM.read(KD_ADDRESS);
+}
+double EEPROM_TC::getMac() {  // Not used?
```
Comment (jgfoster) [inline src/Devices/EEPROM_TC.cpp]: Reference new issue

File: src/Devices/EEPROM_TC.h [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+  // accessor methods
+  double getPH();
+  double getTemp();
+  double getTankID();
+  double getCorrectedTemp();
+  double getKP();
+  double getKI();
+  double getKD();
+  double getMac();  // See a discussion about this function
```
Comment (jgfoster) [inline src/Devices/EEPROM_TC.h]: Reference new issue

**Review by jgfoster — CHANGES_REQUESTED** [review]
Thank you for entering the new issue. I think just adding a link to it in the comments will help us understand the code better when we come back to it later. With that, I think this is very nice. Thanks!

**Review by jgfoster — APPROVED** [review]
LGTM

**Review by prestoncarman — APPROVED** [review]
LGTM

**Pull request merged**